{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "Z-Qpfl-1ii56"
   },
   "source": [
    "# Урок 6\n",
    "\n",
    "## Домашнее задание.\n",
    "Проект по итогам курса\n",
    "По итогу занятий 1, 2, 3 и 4 у вас должна получиться таблица с данными об одном гене. Используя полученную информацию, проанализируйте её, проведите описательную статистику, постройте графики и прочее.\n",
    "\n",
    "Чтобы посмотреть пример работы, обратитесь к разделу 4 на kaggle.com. Полученные результаты предоставляются в формате .ipynb.\n",
    "\n",
    "1. Оценить частоту мутаций в популяциях по выбранному гену.\n",
    "2. Оценить частоту мутаций в разрезе связанных заболеваний."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "I3ZosTT3ii6A"
   },
   "source": [
    "## Задание выполнено командой M4 Team.\n",
    "__Состав команды:__  \n",
    " - Мгер Парунакян\n",
    " - Михаил Демин\n",
    " - Марат Закиров\n",
    " - Марина Дружинина"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "crWKSsLJM2gD"
   },
   "outputs": [],
   "source": [
    "!pip install biopython\n",
    "from Bio import Entrez\n",
    "Entrez.email = \"A.N.Other@example.com\"\n",
    "import pandas as pd\n",
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "from time import sleep\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "executionInfo": {
     "elapsed": 27705,
     "status": "ok",
     "timestamp": 1680881719343,
     "user": {
      "displayName": "Михаил Демин",
      "userId": "14250540086915616791"
     },
     "user_tz": -600
    },
    "id": "HuxegapU5TAJ",
    "outputId": "80fcc827-c84c-4c05-9792-ef53866f5fd5"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Mounted at /content/drive\n"
     ]
    }
   ],
   "source": [
    "from google.colab import drive\n",
    "drive.mount('/content/drive')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "executionInfo": {
     "elapsed": 2129,
     "status": "ok",
     "timestamp": 1680881721463,
     "user": {
      "displayName": "Михаил Демин",
      "userId": "14250540086915616791"
     },
     "user_tz": -600
    },
    "id": "_T5YzuBr1x8F",
    "outputId": "35d6592f-b45d-4186-d22f-83f8ee0df0b0"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/content/drive/MyDrive/Colab Notebooks/GB/19_6_files\n"
     ]
    }
   ],
   "source": [
    "%cd '/content/drive/MyDrive/Colab Notebooks/GB/19_6_files'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "executionInfo": {
     "elapsed": 3726,
     "status": "ok",
     "timestamp": 1680878360030,
     "user": {
      "displayName": "Colab Fold",
      "userId": "04882818707630473374"
     },
     "user_tz": -600
    },
    "id": "pdgXQllIUzp5",
    "outputId": "7a1a95e7-8c5f-4133-d0cc-cec7ed13d2ea"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "96"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Получаем информацию из базы данных SNP для гена FGFR2\n",
    "gene_name = \"FGFR2\"\n",
    "org = \"homo sapiens\"\n",
    "\n",
    "# search ClinVar for variants in the gene\n",
    "try:\n",
    "    handle = Entrez.esearch(db=\"snp\", term=f\"{gene_name}[Gene Name]  AND {org}[Organism] AND (pathogenic[Clinical_Significance] OR likely pathogenic[Clinical_Significance] OR pathogenic likely pathogenic[Clinical_Significance])\", retmax=100)\n",
    "    \n",
    "except urllib.error.HTTPError as e:\n",
    "    if e.code == 429:\n",
    "        print(\"Too many requests. Waiting for 10 seconds...\")\n",
    "        time.sleep(10)\n",
    "        handle = Entrez.esearch(db=\"snp\", term=f\"{gene_name}[Gene Name]  AND {org}[Organism] AND (pathogenic[Clinical_Significance] OR likely pathogenic[Clinical_Significance] OR pathogenic likely pathogenic[Clinical_Significance])\", retmax=100)\n",
    "    else:\n",
    "        raise e\n",
    "record = Entrez.read(handle)\n",
    "handle.close()\n",
    "rs_list = record['IdList']\n",
    "len(rs_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "executionInfo": {
     "elapsed": 20,
     "status": "ok",
     "timestamp": 1680878360030,
     "user": {
      "displayName": "Colab Fold",
      "userId": "04882818707630473374"
     },
     "user_tz": -600
    },
    "id": "dYiuIi3x-Q3I",
    "outputId": "8ef69131-c025-4316-ffa2-4ed2196f2dc3"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['2134256765', '2134256250', '2134253609', '2133823028', '1850314485', '1850299732', '1850289942', '1847510893', '1847174277', '1845559552', '1589828632', '1589722765', '1564919048', '1564875577', '1554930637', '1554928978', '1554928884', '1554927408', '1554917471', '1554914180', '1554907364', '1554907337', '1434545235', '1358919643', '1274989878', '1064796452', '1064796413', '1057520044', '1057520029', '1057520028', '1057520027', '1057519901', '1057519900', '1057519854', '1057519800', '1057519799', '1057519798', '1057519797', '1057519796', '1057519795', '1057519791', '1057519047', '1057519046', '1057519045', '1057519044', '1057519043', '1057519042', '1057519041', '1057519040', '1057519039', '1057519038', '1057519037', '1057519036', '886037837', '879253721', '879253720', '879253719', '879253718', '777169135', '776587763', '387907372', '387906678', '387906677', '387906676', '281865420', '121918510', '121918509', '121918508', '121918507', '121918506', '121918505', '121918504', '121918503', '121918502', '121918501', '121918500', '121918499', '121918498', '121918497', '121918496', '121918495', '121918494', '121918493', '121918492', '121918491', '121918490', '121918489', '121918488', '121918487', '121913478', '121913477', '121913476', '121913475', '121913474', '79184941', '77543610']"
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rs_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "yltiBgXn300c"
   },
   "outputs": [],
   "source": [
    "# Получаем информацию по каждой мутации из списка\n",
    "def get_snp_rs_info(rs_list):\n",
    "  snp_rs_info = []\n",
    "  for k, i in enumerate(rs_list):\n",
    "    print(k)\n",
    "    try:\n",
    "        handle = Entrez.esearch(db=\"snp\", term=\"rs\" + i, format='gb')\n",
    "    except urllib.error.HTTPError as e:\n",
    "        if e.code == 429:\n",
    "            print(\"Too many requests. Waiting for 10 seconds...\")\n",
    "            time.sleep(10)\n",
    "            handle = Entrez.esearch(db=\"snp\", term=\"rs\" + i, format='gb')\n",
    "        else:\n",
    "            raise e\n",
    "    record = Entrez.read(handle)\n",
    "    handle.close()\n",
    "    idlist = record[\"IdList\"]\n",
    "    if len(idlist) > 0:\n",
    "        handle = Entrez.esummary(db=\"snp\", id=\",\".join(idlist))\n",
    "        result = Entrez.read(handle)\n",
    "        handle.close()\n",
    "        snp_info = result['DocumentSummarySet']['DocumentSummary'][0]\n",
    "        snp_rs_info.append({key: snp_info[key] for key in snp_info.keys()})\n",
    "        #snp_rs_info.append({'SNP_ID':f\"rs{snp_info['SNP_ID']}\", 'GENES':snp_info['GENES'], 'ACC':snp_info['ACC'], 'SPDI':snp_info['SPDI'], 'ALLELE':\" > \".join(snp_info['SPDI'].split(':')[2:]), 'CLINICAL_SIGNIFICANCE':snp_info['CLINICAL_SIGNIFICANCE'], 'CLINICAL_SORT':snp_info['CLINICAL_SORT']})\n",
    "  return snp_rs_info\n",
    "  #return result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "executionInfo": {
     "elapsed": 91997,
     "status": "ok",
     "timestamp": 1680878469873,
     "user": {
      "displayName": "Colab Fold",
      "userId": "04882818707630473374"
     },
     "user_tz": -600
    },
    "id": "hwM6npfM380E",
    "outputId": "1338b4d7-a309-4884-fba1-8305e74878bb"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "26\n",
      "27\n",
      "28\n",
      "29\n",
      "30\n",
      "31\n",
      "32\n",
      "33\n",
      "34\n",
      "35\n",
      "36\n",
      "37\n",
      "38\n",
      "39\n",
      "40\n",
      "41\n",
      "42\n",
      "43\n",
      "44\n",
      "45\n",
      "46\n",
      "47\n",
      "48\n",
      "49\n",
      "50\n",
      "51\n",
      "52\n",
      "53\n",
      "54\n",
      "55\n",
      "56\n",
      "57\n",
      "58\n",
      "59\n",
      "60\n",
      "61\n",
      "62\n",
      "63\n",
      "64\n",
      "65\n",
      "66\n",
      "67\n",
      "68\n",
      "69\n",
      "70\n",
      "71\n",
      "72\n",
      "73\n",
      "74\n",
      "75\n",
      "76\n",
      "77\n",
      "78\n",
      "79\n",
      "80\n",
      "81\n",
      "82\n",
      "83\n",
      "84\n",
      "85\n",
      "86\n",
      "87\n",
      "88\n",
      "89\n",
      "90\n",
      "91\n",
      "92\n",
      "93\n",
      "94\n",
      "95\n"
     ]
    }
   ],
   "source": [
    "df_FGFR2_snp = pd.DataFrame(get_snp_rs_info(rs_list))\n",
    "df_FGFR2_snp.to_csv(f'FGFR2_snp.csv', sep=';', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "NKRREtxdljeZ"
   },
   "outputs": [],
   "source": [
    "# Получаем информацию по популяциям\n",
    "# from multiprocessing import Pool\n",
    "\n",
    "def get_population(rs_list):\n",
    "    df_base = pd.DataFrame(columns=['RS_Num', 'Study', 'Population', 'Group', 'Samplesize', 'Ref Allele', 'Alt Allele'])\n",
    "    q = 1\n",
    "    for i in rs_list:\n",
    "        i = \"rs\" + i\n",
    "        print(q, i)\n",
    "        q += 1\n",
    "        url = 'https://www.ncbi.nlm.nih.gov/snp/'+ i\n",
    "\n",
    "        # Send a request to the website and get the HTML content\n",
    "        response = requests.get(url)\n",
    "        html_content = response.content\n",
    "\n",
    "        # Use BeautifulSoup to parse the HTML content\n",
    "        soup = BeautifulSoup(html_content, 'html.parser')\n",
    "\n",
    "        # Find the table containing the variant information\n",
    "        table = soup.find(id = 'dbsnp_freq_datatable')\n",
    "        # Get the rows of the table\n",
    "        try:\n",
    "            rows = table.find_all('tr')\n",
    "            \n",
    "            # Iterate through the rows and extract the table elements\n",
    "            table_elements = []\n",
    "            for row in rows:\n",
    "                columns = row.find_all('td')\n",
    "                row_data = []\n",
    "                for column in columns:\n",
    "                    row_data.append(column.get_text().strip())\n",
    "                table_elements.append(row_data)\n",
    "\n",
    "            # Print the table elements\n",
    "            rows = []\n",
    "            for row in table_elements:\n",
    "                row.insert(0, i)\n",
    "                rows.append(row)\n",
    "            rows.pop(0)\n",
    "            df = pd.DataFrame(rows, columns=['RS_Num', 'Study', 'Population', 'Group', 'Samplesize', 'Ref Allele', 'Alt Allele'])\n",
    "            df_base = pd.concat([df_base, df])\n",
    "        except Exception as ex:\n",
    "            print(ex)\n",
    "            continue\n",
    "    return df_base.reset_index()\n",
    "\n",
    "# if __name__ == '__main__':\n",
    "#   pool = Pool(processes=16)\n",
    "#   results = pool.map(get_population, rs_list)\n",
    "#   pool.close()\n",
    "#   pool.join()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "executionInfo": {
     "elapsed": 217029,
     "status": "ok",
     "timestamp": 1680878686895,
     "user": {
      "displayName": "Colab Fold",
      "userId": "04882818707630473374"
     },
     "user_tz": -600
    },
    "id": "b5ohV8XxPTZY",
    "outputId": "ea00e44e-abcb-446a-87ac-f501e38369cd"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 rs2134256765\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "2 rs2134256250\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "3 rs2134253609\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "4 rs2133823028\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "5 rs1850314485\n",
      "6 rs1850299732\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "7 rs1850289942\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "8 rs1847510893\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "9 rs1847174277\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "10 rs1845559552\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "11 rs1589828632\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "12 rs1589722765\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "13 rs1564919048\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "14 rs1564875577\n",
      "15 rs1554930637\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "16 rs1554928978\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "17 rs1554928884\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "18 rs1554927408\n",
      "19 rs1554917471\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "20 rs1554914180\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "21 rs1554907364\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "22 rs1554907337\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "23 rs1434545235\n",
      "24 rs1358919643\n",
      "25 rs1274989878\n",
      "26 rs1064796452\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "27 rs1064796413\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "28 rs1057520044\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "29 rs1057520029\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "30 rs1057520028\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "31 rs1057520027\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "32 rs1057519901\n",
      "33 rs1057519900\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "34 rs1057519854\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "35 rs1057519800\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "36 rs1057519799\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "37 rs1057519798\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "38 rs1057519797\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "39 rs1057519796\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "40 rs1057519795\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "41 rs1057519791\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "42 rs1057519047\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "43 rs1057519046\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "44 rs1057519045\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "45 rs1057519044\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "46 rs1057519043\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "47 rs1057519042\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "48 rs1057519041\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "49 rs1057519040\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "50 rs1057519039\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "51 rs1057519038\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "52 rs1057519037\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "53 rs1057519036\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "54 rs886037837\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "55 rs879253721\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "56 rs879253720\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "57 rs879253719\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "58 rs879253718\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "59 rs777169135\n",
      "60 rs776587763\n",
      "61 rs387907372\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "62 rs387906678\n",
      "63 rs387906677\n",
      "64 rs387906676\n",
      "65 rs281865420\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "66 rs121918510\n",
      "67 rs121918509\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "68 rs121918508\n",
      "69 rs121918507\n",
      "70 rs121918506\n",
      "71 rs121918505\n",
      "72 rs121918504\n",
      "73 rs121918503\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "74 rs121918502\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "75 rs121918501\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "76 rs121918500\n",
      "77 rs121918499\n",
      "78 rs121918498\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "79 rs121918497\n",
      "80 rs121918496\n",
      "81 rs121918495\n",
      "82 rs121918494\n",
      "83 rs121918493\n",
      "84 rs121918492\n",
      "85 rs121918491\n",
      "86 rs121918490\n",
      "87 rs121918489\n",
      "88 rs121918488\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "89 rs121918487\n",
      "90 rs121913478\n",
      "91 rs121913477\n",
      "92 rs121913476\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "93 rs121913475\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "94 rs121913474\n",
      "'NoneType' object has no attribute 'find_all'\n",
      "95 rs79184941\n",
      "96 rs77543610\n"
     ]
    }
   ],
   "source": [
    "df_FGFR2_pop = get_population(rs_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "k1OseLDGN8CM"
   },
   "outputs": [],
   "source": [
    "df_FGFR2_pop.to_csv('FGFR2_pop.csv', sep=';', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "gahIUKUrg_Bf"
   },
   "outputs": [],
   "source": [
    "# Получаем информацию по болезням связанным с мутацией\n",
    "# from multiprocessing import Pool\n",
    "\n",
    "def get_rs_disease(rs_list):\n",
    "    df_base = pd.DataFrame(columns=['SNP_ID_RS', 'ClinVar_Accession', 'Disease_Names', 'Clinical_Significance', 'Mutation_type'])\n",
    "    q = 1\n",
    "    for i in rs_list:\n",
    "        i = \"rs\" + i\n",
    "        print(q, i)\n",
    "        q += 1\n",
    "        url = 'https://www.ncbi.nlm.nih.gov/snp/'+ i # + '#clinical_significance'\n",
    "\n",
    "        # Send a request to the website and get the HTML content\n",
    "        response = requests.get(url)\n",
    "        html_content = response.content\n",
    "\n",
    "        # Use BeautifulSoup to parse the HTML content\n",
    "        soup = BeautifulSoup(html_content, 'html.parser')\n",
    "\n",
    "        # Find the table containing the variant information\n",
    "        table = soup.find(id = 'clinical_significance_datatable')\n",
    "\n",
    "        # Find the <span> element with the clinical significance value using its XPath\n",
    "        mut_type_dl = soup.find_all('dl', {'class': 'usa-width-one-half'})[1].find_all('dd')[1].find('span')\n",
    "        # dd = consequence_dl[1].find_all('dd')[1].find('span')\n",
    "        # span = dd[1].find('span')\n",
    "        \n",
    "        # Get the rows of the table\n",
    "        try:\n",
    "            rows = table.find_all('tr')\n",
    "            \n",
    "            # Iterate through the rows and extract the table elements\n",
    "            table_elements = []\n",
    "            for row in rows:\n",
    "                columns = row.find_all('td')\n",
    "                row_data = []\n",
    "                for column in columns:\n",
    "                    row_data.append(column.get_text().strip())\n",
    "                table_elements.append(row_data)\n",
    "\n",
    "            # Print the table elements\n",
    "            rows = []\n",
    "            for row in table_elements:\n",
    "                row.insert(0, i)\n",
    "                rows.append(row)\n",
    "            rows.pop(0)\n",
    "            df = pd.DataFrame(rows, columns=['SNP_ID_RS', 'ClinVar_Accession', 'Disease_Names', 'Clinical_Significance'])\n",
    "            mut_type = str(mut_type_dl.text).split(': ')[1]\n",
    "            df['Mutation_type'] = mut_type\n",
    "            df_base = pd.concat([df_base, df])\n",
    "        except Exception as ex:\n",
    "            print(ex)\n",
    "            continue\n",
    "    return df_base.reset_index()\n",
    "\n",
    "# if __name__ == '__main__':\n",
    "#   pool = Pool(processes=16)\n",
    "#   results = pool.map(get_population, rs_list)\n",
    "#   pool.close()\n",
    "#   pool.join()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "executionInfo": {
     "elapsed": 211131,
     "status": "ok",
     "timestamp": 1680880187356,
     "user": {
      "displayName": "Colab Fold",
      "userId": "04882818707630473374"
     },
     "user_tz": -600
    },
    "id": "cWgEAxbLka78",
    "outputId": "b4072908-12bc-456c-9478-99e3f491d5f1"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 rs2134256765\n",
      "2 rs2134256250\n",
      "3 rs2134253609\n",
      "4 rs2133823028\n",
      "5 rs1850314485\n",
      "6 rs1850299732\n",
      "7 rs1850289942\n",
      "8 rs1847510893\n",
      "9 rs1847174277\n",
      "10 rs1845559552\n",
      "11 rs1589828632\n",
      "12 rs1589722765\n",
      "13 rs1564919048\n",
      "14 rs1564875577\n",
      "15 rs1554930637\n",
      "16 rs1554928978\n",
      "17 rs1554928884\n",
      "18 rs1554927408\n",
      "19 rs1554917471\n",
      "20 rs1554914180\n",
      "21 rs1554907364\n",
      "22 rs1554907337\n",
      "23 rs1434545235\n",
      "24 rs1358919643\n",
      "25 rs1274989878\n",
      "26 rs1064796452\n",
      "27 rs1064796413\n",
      "28 rs1057520044\n",
      "29 rs1057520029\n",
      "30 rs1057520028\n",
      "31 rs1057520027\n",
      "32 rs1057519901\n",
      "33 rs1057519900\n",
      "34 rs1057519854\n",
      "35 rs1057519800\n",
      "36 rs1057519799\n",
      "37 rs1057519798\n",
      "38 rs1057519797\n",
      "39 rs1057519796\n",
      "40 rs1057519795\n",
      "41 rs1057519791\n",
      "42 rs1057519047\n",
      "43 rs1057519046\n",
      "44 rs1057519045\n",
      "45 rs1057519044\n",
      "46 rs1057519043\n",
      "47 rs1057519042\n",
      "48 rs1057519041\n",
      "49 rs1057519040\n",
      "50 rs1057519039\n",
      "51 rs1057519038\n",
      "52 rs1057519037\n",
      "53 rs1057519036\n",
      "54 rs886037837\n",
      "55 rs879253721\n",
      "56 rs879253720\n",
      "57 rs879253719\n",
      "58 rs879253718\n",
      "59 rs777169135\n",
      "60 rs776587763\n",
      "61 rs387907372\n",
      "62 rs387906678\n",
      "63 rs387906677\n",
      "64 rs387906676\n",
      "65 rs281865420\n",
      "66 rs121918510\n",
      "67 rs121918509\n",
      "68 rs121918508\n",
      "69 rs121918507\n",
      "70 rs121918506\n",
      "71 rs121918505\n",
      "72 rs121918504\n",
      "73 rs121918503\n",
      "74 rs121918502\n",
      "75 rs121918501\n",
      "76 rs121918500\n",
      "77 rs121918499\n",
      "78 rs121918498\n",
      "79 rs121918497\n",
      "80 rs121918496\n",
      "81 rs121918495\n",
      "82 rs121918494\n",
      "83 rs121918493\n",
      "84 rs121918492\n",
      "85 rs121918491\n",
      "86 rs121918490\n",
      "87 rs121918489\n",
      "88 rs121918488\n",
      "89 rs121918487\n",
      "90 rs121913478\n",
      "91 rs121913477\n",
      "92 rs121913476\n",
      "93 rs121913475\n",
      "94 rs121913474\n",
      "95 rs79184941\n",
      "96 rs77543610\n"
     ]
    }
   ],
   "source": [
    "df_FGFR2_dis = get_rs_disease(rs_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "u-bqbY71ls9H"
   },
   "outputs": [],
   "source": [
    "df_FGFR2_dis.to_csv('FGFR2_dis.csv', sep=';', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "PjHzzeOeDFLQ"
   },
   "outputs": [],
   "source": [
    "# part = 200\n",
    "# parts = int(len(rs_list) / part)\n",
    "# for i in range(int(parts)):\n",
    "#     df = get_population(rs_list[i*part:(i+1)*part])\n",
    "#     if i == parts-1:\n",
    "#         df = get_population(rs_list[(i+1)*part:])\n",
    "#     del df['index']\n",
    "#     df.to_csv(f'df{(i+1)*part}.csv', sep=';')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "8x5UOVD3DFLR"
   },
   "outputs": [],
   "source": [
    "# s=0\n",
    "# part = 200\n",
    "# parts = len(rs_list) / part\n",
    "# for i in range(int(parts)):\n",
    "#     df = get_population(rs_list[s:part])\n",
    "#     s=part\n",
    "#     part+=200\n",
    "#     del df['index']\n",
    "#     df.to_csv(f'df{s}.csv', sep=';')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "awqkmuONDFLS"
   },
   "outputs": [],
   "source": [
    "# # Соединим все файлы в один\n",
    "  \n",
    "# path = \"/content/df\" #hw_6_gene\"  \n",
    "# # Change the directory\n",
    "# os.chdir(path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "rr4jwRoHDFLS"
   },
   "outputs": [],
   "source": [
    "# def read_text_file(file_path,  sep=','):\n",
    "#     with open(file_path, 'r') as f:\n",
    "#         return pd.read_csv(f, sep=sep, index_col= 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "ewgtlOiVDFLT"
   },
   "outputs": [],
   "source": [
    "# # iterate through all file\n",
    "# df = pd.DataFrame(columns=['RS_Num', 'Study', 'Population', 'Group', 'Samplesize', 'Ref Allele', 'Alt Allele'])\n",
    "# for file in os.listdir():\n",
    "#     # Check whether file is in text format or not\n",
    "#     if file.endswith(\".csv\"):\n",
    "#         file_path = f\"{path}//{file}\"\n",
    "#         df_part = read_text_file(file_path, sep=';')\n",
    "#         df = pd.concat([df, df_part])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "fHkGnKKRDFLU"
   },
   "outputs": [],
   "source": [
    "# df_part = read_text_file('df_end.csv', sep=';')\n",
    "# df = pd.concat([df, df_part])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "ZFxB_S8pPT6t"
   },
   "outputs": [],
   "source": [
    "# df_part = read_text_file('/content/drive/MyDrive/Colab_Notebooks/microbio/hw_6/df_all.csv', sep=';')\n",
    "# df = read_text_file('/content/drive/MyDrive/Colab_Notebooks/microbio/hw_6/df_population', sep=';')\n",
    "# df = pd.concat([df, df_part])\n",
    "# df.drop_duplicates(keep='first', inplace=True, ignore_index=True)\n",
    "# df.to_csv('/content/drive/MyDrive/Colab_Notebooks/microbio/hw_6/df_fgfr2_population.csv', sep=';')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 81
    },
    "executionInfo": {
     "elapsed": 81,
     "status": "ok",
     "timestamp": 1680880187365,
     "user": {
      "displayName": "Colab Fold",
      "userId": "04882818707630473374"
     },
     "user_tz": -600
    },
    "id": "9kXmghYF47uN",
    "outputId": "2b93c25a-506a-4aa9-8310-94661ea2863e"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "  <div id=\"df-63dd706a-4ddd-49cc-9335-d2d78b0c353c\">\n",
       "    <div class=\"colab-df-container\">\n",
       "      <div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>RS_Num</th>\n",
       "      <th>Study</th>\n",
       "      <th>Population</th>\n",
       "      <th>Group</th>\n",
       "      <th>Samplesize</th>\n",
       "      <th>Ref Allele</th>\n",
       "      <th>Alt Allele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>rs1850314485</td>\n",
       "      <td>TopMed</td>\n",
       "      <td>Global</td>\n",
       "      <td>Study-wide</td>\n",
       "      <td>264690</td>\n",
       "      <td>C=0.999996</td>\n",
       "      <td>T=0.000004</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>\n",
       "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-63dd706a-4ddd-49cc-9335-d2d78b0c353c')\"\n",
       "              title=\"Convert this dataframe to an interactive table.\"\n",
       "              style=\"display:none;\">\n",
       "        \n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "       width=\"24px\">\n",
       "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
       "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
       "  </svg>\n",
       "      </button>\n",
       "      \n",
       "  <style>\n",
       "    .colab-df-container {\n",
       "      display:flex;\n",
       "      flex-wrap:wrap;\n",
       "      gap: 12px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert {\n",
       "      background-color: #E8F0FE;\n",
       "      border: none;\n",
       "      border-radius: 50%;\n",
       "      cursor: pointer;\n",
       "      display: none;\n",
       "      fill: #1967D2;\n",
       "      height: 32px;\n",
       "      padding: 0 0 0 0;\n",
       "      width: 32px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert:hover {\n",
       "      background-color: #E2EBFA;\n",
       "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "      fill: #174EA6;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert {\n",
       "      background-color: #3B4455;\n",
       "      fill: #D2E3FC;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert:hover {\n",
       "      background-color: #434B5C;\n",
       "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "      fill: #FFFFFF;\n",
       "    }\n",
       "  </style>\n",
       "\n",
       "      <script>\n",
       "        const buttonEl =\n",
       "          document.querySelector('#df-63dd706a-4ddd-49cc-9335-d2d78b0c353c button.colab-df-convert');\n",
       "        buttonEl.style.display =\n",
       "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "        async function convertToInteractive(key) {\n",
       "          const element = document.querySelector('#df-63dd706a-4ddd-49cc-9335-d2d78b0c353c');\n",
       "          const dataTable =\n",
       "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
       "                                                     [key], {});\n",
       "          if (!dataTable) return;\n",
       "\n",
       "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
       "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
       "            + ' to learn more about interactive tables.';\n",
       "          element.innerHTML = '';\n",
       "          dataTable['output_type'] = 'display_data';\n",
       "          await google.colab.output.renderOutput(dataTable, element);\n",
       "          const docLink = document.createElement('div');\n",
       "          docLink.innerHTML = docLinkHtml;\n",
       "          element.appendChild(docLink);\n",
       "        }\n",
       "      </script>\n",
       "    </div>\n",
       "  </div>\n",
       "  "
      ],
      "text/plain": [
       "   index        RS_Num   Study Population       Group  Samplesize  Ref Allele  \\\n",
       "0      0  rs1850314485  TopMed     Global  Study-wide      264690  C=0.999996   \n",
       "\n",
       "   Alt Allele  \n",
       "0  T=0.000004  "
      ]
     },
     "execution_count": 120,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_pop = pd.read_csv('FGFR2_pop.csv', sep=';')\n",
    "df_pop.head(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 226
    },
    "executionInfo": {
     "elapsed": 78,
     "status": "ok",
     "timestamp": 1680880187366,
     "user": {
      "displayName": "Colab Fold",
      "userId": "04882818707630473374"
     },
     "user_tz": -600
    },
    "id": "dguOzHB7XQvA",
    "outputId": "1101bb39-1955-4a02-dedb-e6dd084a0a43"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "  <div id=\"df-5b30d1ac-5f61-4e5a-b0a8-964f37859122\">\n",
       "    <div class=\"colab-df-container\">\n",
       "      <div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP_ID</th>\n",
       "      <th>ALLELE_ORIGIN</th>\n",
       "      <th>GLOBAL_MAFS</th>\n",
       "      <th>GLOBAL_POPULATION</th>\n",
       "      <th>GLOBAL_SAMPLESIZE</th>\n",
       "      <th>SUSPECTED</th>\n",
       "      <th>CLINICAL_SIGNIFICANCE</th>\n",
       "      <th>GENES</th>\n",
       "      <th>ACC</th>\n",
       "      <th>CHR</th>\n",
       "      <th>...</th>\n",
       "      <th>ALLELE</th>\n",
       "      <th>SNP_CLASS</th>\n",
       "      <th>CHRPOS</th>\n",
       "      <th>CHRPOS_PREV_ASSM</th>\n",
       "      <th>TEXT</th>\n",
       "      <th>SNP_ID_SORT</th>\n",
       "      <th>CLINICAL_SORT</th>\n",
       "      <th>CITED_SORT</th>\n",
       "      <th>CHRPOS_SORT</th>\n",
       "      <th>MERGED_SORT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2134256765</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>likely-pathogenic</td>\n",
       "      <td>[{'NAME': 'FGFR2', 'GENE_ID': '2263'}]</td>\n",
       "      <td>NC_000010.11</td>\n",
       "      <td>10</td>\n",
       "      <td>...</td>\n",
       "      <td>N</td>\n",
       "      <td>ins</td>\n",
       "      <td>10:121517381</td>\n",
       "      <td>10:123276896</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2134256765</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>121517381</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 31 columns</p>\n",
       "</div>\n",
       "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-5b30d1ac-5f61-4e5a-b0a8-964f37859122')\"\n",
       "              title=\"Convert this dataframe to an interactive table.\"\n",
       "              style=\"display:none;\">\n",
       "        \n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "       width=\"24px\">\n",
       "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
       "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
       "  </svg>\n",
       "      </button>\n",
       "      \n",
       "  <style>\n",
       "    .colab-df-container {\n",
       "      display:flex;\n",
       "      flex-wrap:wrap;\n",
       "      gap: 12px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert {\n",
       "      background-color: #E8F0FE;\n",
       "      border: none;\n",
       "      border-radius: 50%;\n",
       "      cursor: pointer;\n",
       "      display: none;\n",
       "      fill: #1967D2;\n",
       "      height: 32px;\n",
       "      padding: 0 0 0 0;\n",
       "      width: 32px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert:hover {\n",
       "      background-color: #E2EBFA;\n",
       "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "      fill: #174EA6;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert {\n",
       "      background-color: #3B4455;\n",
       "      fill: #D2E3FC;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert:hover {\n",
       "      background-color: #434B5C;\n",
       "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "      fill: #FFFFFF;\n",
       "    }\n",
       "  </style>\n",
       "\n",
       "      <script>\n",
       "        const buttonEl =\n",
       "          document.querySelector('#df-5b30d1ac-5f61-4e5a-b0a8-964f37859122 button.colab-df-convert');\n",
       "        buttonEl.style.display =\n",
       "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "        async function convertToInteractive(key) {\n",
       "          const element = document.querySelector('#df-5b30d1ac-5f61-4e5a-b0a8-964f37859122');\n",
       "          const dataTable =\n",
       "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
       "                                                     [key], {});\n",
       "          if (!dataTable) return;\n",
       "\n",
       "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
       "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
       "            + ' to learn more about interactive tables.';\n",
       "          element.innerHTML = '';\n",
       "          dataTable['output_type'] = 'display_data';\n",
       "          await google.colab.output.renderOutput(dataTable, element);\n",
       "          const docLink = document.createElement('div');\n",
       "          docLink.innerHTML = docLinkHtml;\n",
       "          element.appendChild(docLink);\n",
       "        }\n",
       "      </script>\n",
       "    </div>\n",
       "  </div>\n",
       "  "
      ],
      "text/plain": [
       "       SNP_ID  ALLELE_ORIGIN GLOBAL_MAFS  GLOBAL_POPULATION  \\\n",
       "0  2134256765            NaN          []                NaN   \n",
       "\n",
       "   GLOBAL_SAMPLESIZE  SUSPECTED CLINICAL_SIGNIFICANCE  \\\n",
       "0                  0        NaN     likely-pathogenic   \n",
       "\n",
       "                                    GENES           ACC  CHR  ... ALLELE  \\\n",
       "0  [{'NAME': 'FGFR2', 'GENE_ID': '2263'}]  NC_000010.11   10  ...      N   \n",
       "\n",
       "  SNP_CLASS        CHRPOS CHRPOS_PREV_ASSM TEXT  SNP_ID_SORT  CLINICAL_SORT  \\\n",
       "0       ins  10:121517381     10:123276896  NaN   2134256765              1   \n",
       "\n",
       "   CITED_SORT CHRPOS_SORT MERGED_SORT  \n",
       "0         NaN   121517381           0  \n",
       "\n",
       "[1 rows x 31 columns]"
      ]
     },
     "execution_count": 121,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_all = pd.read_csv('FGFR2_snp.csv', sep=';')\n",
    "df_all.head(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 81
    },
    "executionInfo": {
     "elapsed": 73,
     "status": "ok",
     "timestamp": 1680880187367,
     "user": {
      "displayName": "Colab Fold",
      "userId": "04882818707630473374"
     },
     "user_tz": -600
    },
    "id": "KZf0XSNVrZok",
    "outputId": "e0be9ad7-f77f-460a-ae01-2b6b9cd62146"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "  <div id=\"df-29b6dc92-cc21-4efd-be51-914882fb8198\">\n",
       "    <div class=\"colab-df-container\">\n",
       "      <div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>SNP_ID_RS</th>\n",
       "      <th>ClinVar_Accession</th>\n",
       "      <th>Disease_Names</th>\n",
       "      <th>Clinical_Significance</th>\n",
       "      <th>Mutation_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>rs2134256765</td>\n",
       "      <td>RCV002224437.1</td>\n",
       "      <td>not provided</td>\n",
       "      <td>Likely-Pathogenic</td>\n",
       "      <td>Inframe Indel</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>\n",
       "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-29b6dc92-cc21-4efd-be51-914882fb8198')\"\n",
       "              title=\"Convert this dataframe to an interactive table.\"\n",
       "              style=\"display:none;\">\n",
       "        \n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "       width=\"24px\">\n",
       "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
       "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
       "  </svg>\n",
       "      </button>\n",
       "      \n",
       "  <style>\n",
       "    .colab-df-container {\n",
       "      display:flex;\n",
       "      flex-wrap:wrap;\n",
       "      gap: 12px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert {\n",
       "      background-color: #E8F0FE;\n",
       "      border: none;\n",
       "      border-radius: 50%;\n",
       "      cursor: pointer;\n",
       "      display: none;\n",
       "      fill: #1967D2;\n",
       "      height: 32px;\n",
       "      padding: 0 0 0 0;\n",
       "      width: 32px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert:hover {\n",
       "      background-color: #E2EBFA;\n",
       "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "      fill: #174EA6;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert {\n",
       "      background-color: #3B4455;\n",
       "      fill: #D2E3FC;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert:hover {\n",
       "      background-color: #434B5C;\n",
       "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "      fill: #FFFFFF;\n",
       "    }\n",
       "  </style>\n",
       "\n",
       "      <script>\n",
       "        const buttonEl =\n",
       "          document.querySelector('#df-29b6dc92-cc21-4efd-be51-914882fb8198 button.colab-df-convert');\n",
       "        buttonEl.style.display =\n",
       "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "        async function convertToInteractive(key) {\n",
       "          const element = document.querySelector('#df-29b6dc92-cc21-4efd-be51-914882fb8198');\n",
       "          const dataTable =\n",
       "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
       "                                                     [key], {});\n",
       "          if (!dataTable) return;\n",
       "\n",
       "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
       "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
       "            + ' to learn more about interactive tables.';\n",
       "          element.innerHTML = '';\n",
       "          dataTable['output_type'] = 'display_data';\n",
       "          await google.colab.output.renderOutput(dataTable, element);\n",
       "          const docLink = document.createElement('div');\n",
       "          docLink.innerHTML = docLinkHtml;\n",
       "          element.appendChild(docLink);\n",
       "        }\n",
       "      </script>\n",
       "    </div>\n",
       "  </div>\n",
       "  "
      ],
      "text/plain": [
       "   index     SNP_ID_RS ClinVar_Accession Disease_Names Clinical_Significance  \\\n",
       "0      0  rs2134256765    RCV002224437.1  not provided     Likely-Pathogenic   \n",
       "\n",
       "   Mutation_type  \n",
       "0  Inframe Indel  "
      ]
     },
     "execution_count": 122,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_dis = pd.read_csv('FGFR2_dis.csv', sep=';')\n",
    "df_dis.head(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 81
    },
    "executionInfo": {
     "elapsed": 72,
     "status": "ok",
     "timestamp": 1680880187368,
     "user": {
      "displayName": "Colab Fold",
      "userId": "04882818707630473374"
     },
     "user_tz": -600
    },
    "id": "KWSlA-1wX5IJ",
    "outputId": "8a397a7b-0542-4213-a465-1c7e2f7a0472"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "  <div id=\"df-753c07e0-e2ee-4d65-92e9-e65b7edfdfb8\">\n",
       "    <div class=\"colab-df-container\">\n",
       "      <div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP_ID_RS</th>\n",
       "      <th>Study</th>\n",
       "      <th>Population</th>\n",
       "      <th>Group</th>\n",
       "      <th>Samplesize</th>\n",
       "      <th>Ref Allele</th>\n",
       "      <th>Alt Allele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs1850314485</td>\n",
       "      <td>TopMed</td>\n",
       "      <td>Global</td>\n",
       "      <td>Study-wide</td>\n",
       "      <td>264690</td>\n",
       "      <td>C=0.999996</td>\n",
       "      <td>T=0.000004</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>\n",
       "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-753c07e0-e2ee-4d65-92e9-e65b7edfdfb8')\"\n",
       "              title=\"Convert this dataframe to an interactive table.\"\n",
       "              style=\"display:none;\">\n",
       "        \n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "       width=\"24px\">\n",
       "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
       "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
       "  </svg>\n",
       "      </button>\n",
       "      \n",
       "  <style>\n",
       "    .colab-df-container {\n",
       "      display:flex;\n",
       "      flex-wrap:wrap;\n",
       "      gap: 12px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert {\n",
       "      background-color: #E8F0FE;\n",
       "      border: none;\n",
       "      border-radius: 50%;\n",
       "      cursor: pointer;\n",
       "      display: none;\n",
       "      fill: #1967D2;\n",
       "      height: 32px;\n",
       "      padding: 0 0 0 0;\n",
       "      width: 32px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert:hover {\n",
       "      background-color: #E2EBFA;\n",
       "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "      fill: #174EA6;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert {\n",
       "      background-color: #3B4455;\n",
       "      fill: #D2E3FC;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert:hover {\n",
       "      background-color: #434B5C;\n",
       "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "      fill: #FFFFFF;\n",
       "    }\n",
       "  </style>\n",
       "\n",
       "      <script>\n",
       "        const buttonEl =\n",
       "          document.querySelector('#df-753c07e0-e2ee-4d65-92e9-e65b7edfdfb8 button.colab-df-convert');\n",
       "        buttonEl.style.display =\n",
       "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "        async function convertToInteractive(key) {\n",
       "          const element = document.querySelector('#df-753c07e0-e2ee-4d65-92e9-e65b7edfdfb8');\n",
       "          const dataTable =\n",
       "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
       "                                                     [key], {});\n",
       "          if (!dataTable) return;\n",
       "\n",
       "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
       "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
       "            + ' to learn more about interactive tables.';\n",
       "          element.innerHTML = '';\n",
       "          dataTable['output_type'] = 'display_data';\n",
       "          await google.colab.output.renderOutput(dataTable, element);\n",
       "          const docLink = document.createElement('div');\n",
       "          docLink.innerHTML = docLinkHtml;\n",
       "          element.appendChild(docLink);\n",
       "        }\n",
       "      </script>\n",
       "    </div>\n",
       "  </div>\n",
       "  "
      ],
      "text/plain": [
       "      SNP_ID_RS   Study Population       Group  Samplesize  Ref Allele  \\\n",
       "0  rs1850314485  TopMed     Global  Study-wide      264690  C=0.999996   \n",
       "\n",
       "   Alt Allele  \n",
       "0  T=0.000004  "
      ]
     },
     "execution_count": 123,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "del df_pop['index']\n",
    "df_pop.rename(columns = {'RS_Num':'SNP_ID_RS'}, inplace = True )\n",
    "df_pop.head(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 226
    },
    "executionInfo": {
     "elapsed": 71,
     "status": "ok",
     "timestamp": 1680880187369,
     "user": {
      "displayName": "Colab Fold",
      "userId": "04882818707630473374"
     },
     "user_tz": -600
    },
    "id": "m9xMAxgUYV_N",
    "outputId": "0b91dba5-b967-4293-c700-626dda45c523"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "  <div id=\"df-63c87423-cd2d-4a0c-a569-f3a3cc47f4e5\">\n",
       "    <div class=\"colab-df-container\">\n",
       "      <div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP_ID</th>\n",
       "      <th>ALLELE_ORIGIN</th>\n",
       "      <th>GLOBAL_MAFS</th>\n",
       "      <th>GLOBAL_POPULATION</th>\n",
       "      <th>GLOBAL_SAMPLESIZE</th>\n",
       "      <th>SUSPECTED</th>\n",
       "      <th>CLINICAL_SIGNIFICANCE</th>\n",
       "      <th>GENES</th>\n",
       "      <th>ACC</th>\n",
       "      <th>CHR</th>\n",
       "      <th>...</th>\n",
       "      <th>SNP_CLASS</th>\n",
       "      <th>CHRPOS</th>\n",
       "      <th>CHRPOS_PREV_ASSM</th>\n",
       "      <th>TEXT</th>\n",
       "      <th>SNP_ID_SORT</th>\n",
       "      <th>CLINICAL_SORT</th>\n",
       "      <th>CITED_SORT</th>\n",
       "      <th>CHRPOS_SORT</th>\n",
       "      <th>MERGED_SORT</th>\n",
       "      <th>SNP_ID_RS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2134256765</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>likely-pathogenic</td>\n",
       "      <td>[{'NAME': 'FGFR2', 'GENE_ID': '2263'}]</td>\n",
       "      <td>NC_000010.11</td>\n",
       "      <td>10</td>\n",
       "      <td>...</td>\n",
       "      <td>ins</td>\n",
       "      <td>10:121517381</td>\n",
       "      <td>10:123276896</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2134256765</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>121517381</td>\n",
       "      <td>0</td>\n",
       "      <td>rs2134256765</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 32 columns</p>\n",
       "</div>\n",
       "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-63c87423-cd2d-4a0c-a569-f3a3cc47f4e5')\"\n",
       "              title=\"Convert this dataframe to an interactive table.\"\n",
       "              style=\"display:none;\">\n",
       "        \n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "       width=\"24px\">\n",
       "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
       "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
       "  </svg>\n",
       "      </button>\n",
       "      \n",
       "  <style>\n",
       "    .colab-df-container {\n",
       "      display:flex;\n",
       "      flex-wrap:wrap;\n",
       "      gap: 12px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert {\n",
       "      background-color: #E8F0FE;\n",
       "      border: none;\n",
       "      border-radius: 50%;\n",
       "      cursor: pointer;\n",
       "      display: none;\n",
       "      fill: #1967D2;\n",
       "      height: 32px;\n",
       "      padding: 0 0 0 0;\n",
       "      width: 32px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert:hover {\n",
       "      background-color: #E2EBFA;\n",
       "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "      fill: #174EA6;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert {\n",
       "      background-color: #3B4455;\n",
       "      fill: #D2E3FC;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert:hover {\n",
       "      background-color: #434B5C;\n",
       "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "      fill: #FFFFFF;\n",
       "    }\n",
       "  </style>\n",
       "\n",
       "      <script>\n",
       "        const buttonEl =\n",
       "          document.querySelector('#df-63c87423-cd2d-4a0c-a569-f3a3cc47f4e5 button.colab-df-convert');\n",
       "        buttonEl.style.display =\n",
       "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "        async function convertToInteractive(key) {\n",
       "          const element = document.querySelector('#df-63c87423-cd2d-4a0c-a569-f3a3cc47f4e5');\n",
       "          const dataTable =\n",
       "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
       "                                                     [key], {});\n",
       "          if (!dataTable) return;\n",
       "\n",
       "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
       "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
       "            + ' to learn more about interactive tables.';\n",
       "          element.innerHTML = '';\n",
       "          dataTable['output_type'] = 'display_data';\n",
       "          await google.colab.output.renderOutput(dataTable, element);\n",
       "          const docLink = document.createElement('div');\n",
       "          docLink.innerHTML = docLinkHtml;\n",
       "          element.appendChild(docLink);\n",
       "        }\n",
       "      </script>\n",
       "    </div>\n",
       "  </div>\n",
       "  "
      ],
      "text/plain": [
       "       SNP_ID  ALLELE_ORIGIN GLOBAL_MAFS  GLOBAL_POPULATION  \\\n",
       "0  2134256765            NaN          []                NaN   \n",
       "\n",
       "   GLOBAL_SAMPLESIZE  SUSPECTED CLINICAL_SIGNIFICANCE  \\\n",
       "0                  0        NaN     likely-pathogenic   \n",
       "\n",
       "                                    GENES           ACC  CHR  ... SNP_CLASS  \\\n",
       "0  [{'NAME': 'FGFR2', 'GENE_ID': '2263'}]  NC_000010.11   10  ...       ins   \n",
       "\n",
       "         CHRPOS CHRPOS_PREV_ASSM TEXT SNP_ID_SORT  CLINICAL_SORT  CITED_SORT  \\\n",
       "0  10:121517381     10:123276896  NaN  2134256765              1         NaN   \n",
       "\n",
       "   CHRPOS_SORT MERGED_SORT     SNP_ID_RS  \n",
       "0    121517381           0  rs2134256765  \n",
       "\n",
       "[1 rows x 32 columns]"
      ]
     },
     "execution_count": 124,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_all['SNP_ID_RS'] = df_all['SNP_ID'].apply(lambda x: 'rs' + str(x))\n",
    "df_all.head(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 174
    },
    "executionInfo": {
     "elapsed": 69,
     "status": "ok",
     "timestamp": 1680880187370,
     "user": {
      "displayName": "Colab Fold",
      "userId": "04882818707630473374"
     },
     "user_tz": -600
    },
    "id": "UTIPyzSYaUYU",
    "outputId": "6e955dc1-e973-4455-d039-e7b7abdd4af1"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "  <div id=\"df-e6b922e4-b4ce-491a-b914-5f1bf4a9a015\">\n",
       "    <div class=\"colab-df-container\">\n",
       "      <div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>SNP_ID_RS</th>\n",
       "      <th>ClinVar_Accession</th>\n",
       "      <th>Disease_Names</th>\n",
       "      <th>Clinical_Significance</th>\n",
       "      <th>Mutation_type</th>\n",
       "      <th>SNP_ID</th>\n",
       "      <th>ALLELE_ORIGIN</th>\n",
       "      <th>GLOBAL_MAFS</th>\n",
       "      <th>GLOBAL_POPULATION</th>\n",
       "      <th>...</th>\n",
       "      <th>ALLELE</th>\n",
       "      <th>SNP_CLASS</th>\n",
       "      <th>CHRPOS</th>\n",
       "      <th>CHRPOS_PREV_ASSM</th>\n",
       "      <th>TEXT</th>\n",
       "      <th>SNP_ID_SORT</th>\n",
       "      <th>CLINICAL_SORT</th>\n",
       "      <th>CITED_SORT</th>\n",
       "      <th>CHRPOS_SORT</th>\n",
       "      <th>MERGED_SORT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>rs2134256765</td>\n",
       "      <td>RCV002224437.1</td>\n",
       "      <td>not provided</td>\n",
       "      <td>Likely-Pathogenic</td>\n",
       "      <td>Inframe Indel</td>\n",
       "      <td>2134256765</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>N</td>\n",
       "      <td>ins</td>\n",
       "      <td>10:121517381</td>\n",
       "      <td>10:123276896</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2134256765</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>121517381</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 37 columns</p>\n",
       "</div>\n",
       "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-e6b922e4-b4ce-491a-b914-5f1bf4a9a015')\"\n",
       "              title=\"Convert this dataframe to an interactive table.\"\n",
       "              style=\"display:none;\">\n",
       "        \n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "       width=\"24px\">\n",
       "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
       "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
       "  </svg>\n",
       "      </button>\n",
       "      \n",
       "  <style>\n",
       "    .colab-df-container {\n",
       "      display:flex;\n",
       "      flex-wrap:wrap;\n",
       "      gap: 12px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert {\n",
       "      background-color: #E8F0FE;\n",
       "      border: none;\n",
       "      border-radius: 50%;\n",
       "      cursor: pointer;\n",
       "      display: none;\n",
       "      fill: #1967D2;\n",
       "      height: 32px;\n",
       "      padding: 0 0 0 0;\n",
       "      width: 32px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert:hover {\n",
       "      background-color: #E2EBFA;\n",
       "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "      fill: #174EA6;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert {\n",
       "      background-color: #3B4455;\n",
       "      fill: #D2E3FC;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert:hover {\n",
       "      background-color: #434B5C;\n",
       "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "      fill: #FFFFFF;\n",
       "    }\n",
       "  </style>\n",
       "\n",
       "      <script>\n",
       "        const buttonEl =\n",
       "          document.querySelector('#df-e6b922e4-b4ce-491a-b914-5f1bf4a9a015 button.colab-df-convert');\n",
       "        buttonEl.style.display =\n",
       "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "        async function convertToInteractive(key) {\n",
       "          const element = document.querySelector('#df-e6b922e4-b4ce-491a-b914-5f1bf4a9a015');\n",
       "          const dataTable =\n",
       "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
       "                                                     [key], {});\n",
       "          if (!dataTable) return;\n",
       "\n",
       "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
       "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
       "            + ' to learn more about interactive tables.';\n",
       "          element.innerHTML = '';\n",
       "          dataTable['output_type'] = 'display_data';\n",
       "          await google.colab.output.renderOutput(dataTable, element);\n",
       "          const docLink = document.createElement('div');\n",
       "          docLink.innerHTML = docLinkHtml;\n",
       "          element.appendChild(docLink);\n",
       "        }\n",
       "      </script>\n",
       "    </div>\n",
       "  </div>\n",
       "  "
      ],
      "text/plain": [
       "   index     SNP_ID_RS ClinVar_Accession Disease_Names Clinical_Significance  \\\n",
       "0      0  rs2134256765    RCV002224437.1  not provided     Likely-Pathogenic   \n",
       "\n",
       "   Mutation_type      SNP_ID  ALLELE_ORIGIN GLOBAL_MAFS  GLOBAL_POPULATION  \\\n",
       "0  Inframe Indel  2134256765            NaN          []                NaN   \n",
       "\n",
       "   ...  ALLELE  SNP_CLASS        CHRPOS CHRPOS_PREV_ASSM TEXT  SNP_ID_SORT  \\\n",
       "0  ...       N        ins  10:121517381     10:123276896  NaN   2134256765   \n",
       "\n",
       "  CLINICAL_SORT CITED_SORT CHRPOS_SORT MERGED_SORT  \n",
       "0             1        NaN   121517381           0  \n",
       "\n",
       "[1 rows x 37 columns]"
      ]
     },
     "execution_count": 125,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_2 = df_dis.merge(df_all, how='left', on='SNP_ID_RS')\n",
    "df_2.head(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "n4mKhuZ2amgR"
   },
   "outputs": [],
   "source": [
    "df_2.to_parquet('FGFR2_disease_dataset.parquet.gzip', compression='gzip')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 191
    },
    "executionInfo": {
     "elapsed": 66,
     "status": "ok",
     "timestamp": 1680880187371,
     "user": {
      "displayName": "Colab Fold",
      "userId": "04882818707630473374"
     },
     "user_tz": -600
    },
    "id": "tgQZXBlrDB-4",
    "outputId": "78b01c96-ca22-4437-8439-ba21fd1a5d26"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "  <div id=\"df-21bd1230-2da0-4335-9278-305db1157f02\">\n",
       "    <div class=\"colab-df-container\">\n",
       "      <div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP_ID_RS</th>\n",
       "      <th>Study</th>\n",
       "      <th>Population</th>\n",
       "      <th>Group</th>\n",
       "      <th>Samplesize</th>\n",
       "      <th>Ref Allele</th>\n",
       "      <th>Alt Allele</th>\n",
       "      <th>SNP_ID</th>\n",
       "      <th>ALLELE_ORIGIN</th>\n",
       "      <th>GLOBAL_MAFS</th>\n",
       "      <th>...</th>\n",
       "      <th>ALLELE</th>\n",
       "      <th>SNP_CLASS</th>\n",
       "      <th>CHRPOS</th>\n",
       "      <th>CHRPOS_PREV_ASSM</th>\n",
       "      <th>TEXT</th>\n",
       "      <th>SNP_ID_SORT</th>\n",
       "      <th>CLINICAL_SORT</th>\n",
       "      <th>CITED_SORT</th>\n",
       "      <th>CHRPOS_SORT</th>\n",
       "      <th>MERGED_SORT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs2134256765</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2134256765</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>N</td>\n",
       "      <td>ins</td>\n",
       "      <td>10:121517381</td>\n",
       "      <td>10:123276896</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2134256765</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>121517381</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 38 columns</p>\n",
       "</div>\n",
       "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-21bd1230-2da0-4335-9278-305db1157f02')\"\n",
       "              title=\"Convert this dataframe to an interactive table.\"\n",
       "              style=\"display:none;\">\n",
       "        \n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "       width=\"24px\">\n",
       "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
       "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
       "  </svg>\n",
       "      </button>\n",
       "      \n",
       "  <style>\n",
       "    .colab-df-container {\n",
       "      display:flex;\n",
       "      flex-wrap:wrap;\n",
       "      gap: 12px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert {\n",
       "      background-color: #E8F0FE;\n",
       "      border: none;\n",
       "      border-radius: 50%;\n",
       "      cursor: pointer;\n",
       "      display: none;\n",
       "      fill: #1967D2;\n",
       "      height: 32px;\n",
       "      padding: 0 0 0 0;\n",
       "      width: 32px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert:hover {\n",
       "      background-color: #E2EBFA;\n",
       "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "      fill: #174EA6;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert {\n",
       "      background-color: #3B4455;\n",
       "      fill: #D2E3FC;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert:hover {\n",
       "      background-color: #434B5C;\n",
       "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "      fill: #FFFFFF;\n",
       "    }\n",
       "  </style>\n",
       "\n",
       "      <script>\n",
       "        const buttonEl =\n",
       "          document.querySelector('#df-21bd1230-2da0-4335-9278-305db1157f02 button.colab-df-convert');\n",
       "        buttonEl.style.display =\n",
       "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "        async function convertToInteractive(key) {\n",
       "          const element = document.querySelector('#df-21bd1230-2da0-4335-9278-305db1157f02');\n",
       "          const dataTable =\n",
       "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
       "                                                     [key], {});\n",
       "          if (!dataTable) return;\n",
       "\n",
       "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
       "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
       "            + ' to learn more about interactive tables.';\n",
       "          element.innerHTML = '';\n",
       "          dataTable['output_type'] = 'display_data';\n",
       "          await google.colab.output.renderOutput(dataTable, element);\n",
       "          const docLink = document.createElement('div');\n",
       "          docLink.innerHTML = docLinkHtml;\n",
       "          element.appendChild(docLink);\n",
       "        }\n",
       "      </script>\n",
       "    </div>\n",
       "  </div>\n",
       "  "
      ],
      "text/plain": [
       "      SNP_ID_RS Study Population Group  Samplesize Ref Allele Alt Allele  \\\n",
       "0  rs2134256765   NaN        NaN   NaN         NaN        NaN        NaN   \n",
       "\n",
       "       SNP_ID  ALLELE_ORIGIN GLOBAL_MAFS  ...  ALLELE  SNP_CLASS  \\\n",
       "0  2134256765            NaN          []  ...       N        ins   \n",
       "\n",
       "         CHRPOS CHRPOS_PREV_ASSM TEXT SNP_ID_SORT  CLINICAL_SORT CITED_SORT  \\\n",
       "0  10:121517381     10:123276896  NaN  2134256765              1        NaN   \n",
       "\n",
       "  CHRPOS_SORT MERGED_SORT  \n",
       "0   121517381           0  \n",
       "\n",
       "[1 rows x 38 columns]"
      ]
     },
     "execution_count": 127,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_3 = df_pop.merge(df_all, how='right', on='SNP_ID_RS')\n",
    "df_3.head(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "XxcnvB8D89Wj"
   },
   "outputs": [],
   "source": [
    "df_3.to_parquet('FGFR2_population_dataset.parquet.gzip', compression='gzip')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "Xgn6jO3xHLyN"
   },
   "outputs": [],
   "source": [
    "# df2.to_csv(f'/content/drive/MyDrive/Colab_Notebooks/microbio/hw_6/M4team_dataset.csv', sep=';')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "lXqVJutGIvMh"
   },
   "outputs": [],
   "source": [
    "# !zip -r /content/drive/MyDrive/Colab_Notebooks/microbio/hw_6/M4team_dataset.csv /content/M4team_dataset.zip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "DkskEmwaKJuU"
   },
   "outputs": [],
   "source": [
    "# df2.to_parquet('/content/drive/MyDrive/Colab_Notebooks/microbio/hw_6/M4team_dataset.parquet.gzip', compression='gzip')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "qr2zv1TLKs_D"
   },
   "outputs": [],
   "source": [
    "df2 = pd.read_parquet('FGFR2_disease_dataset.parquet.gzip')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "ropDadcjE1rJ"
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "colab": {
   "provenance": []
  },
  "gpuClass": "standard",
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.16"
  },
  "nbTranslate": {
   "displayLangs": [
    "*"
   ],
   "hotkey": "alt-t",
   "langInMainMenu": true,
   "sourceLang": "en",
   "targetLang": "fr",
   "useGoogleTranslate": true
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": false,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
